olive/passes/pytorch/torch_trt_conversion.py

```diff
@@ -0,0 +1,177 @@
+        seqlen = seqlens[model_type]
+
+        # load model
+        pytorch_model = model.load_model()
+        # we will update the model inplace
+        # since the models are large, it is expensive to copy and maintain two copies
+        # set model.model to None so that the input model doesn't use this same model object when it is loaded
+        model.model = None
+        # alternative is to copy the model and use the copy
+        # pytorch_model = copy.deepcopy(model.model)
+        pytorch_model.eval()
+        # move model to device
+        pytorch_model.to(device=device)
+        # convert model to fp16 if needed
+        if config["float16"]:
+            pytorch_model = pytorch_model.to(dtype=torch.float16)
+        # disable cache
+        use_cache = pytorch_model.config.use_cache
+        pytorch_model.config.use_cache = False
+
+        # get module list of layers
+        layers = get_layers(pytorch_model, model_type)
+
+        # get layer information
+        min_layer, max_layer = validate_min_max_layers(config["min_layer"], config["max_layer"], len(layers))
+        layer_name_filter = config["layer_name_filter"] or []
+        if isinstance(layer_name_filter, str):
+            layer_name_filter = [layer_name_filter]
+        # layer information storage
+        layer_info = {}
+        # loop over layers
+        for i in range(min_layer, max_layer):
+            layer = layers[i]
+            layer_info[i] = {"submodules": {}, "handles": [], "input_shapes": {}}
+
+            # get list of submodules in layer
+            layer_info[i]["submodules"] = get_layer_submodules(
+                layer, submodule_types=[torch.nn.Linear], layer_name_filter=layer_name_filter
+            )
+
+            # add forward hook to submodules in layer
+            def get_handler(layer_idx, submodule_name):
+                def handler(_, input, output):
+                    layer_info[layer_idx]["input_shapes"][submodule_name] = input[0].shape
+
+                return handler
+
+            for name, submodule in layer_info[i]["submodules"].items():
+                layer_info[i]["handles"].append(submodule.register_forward_hook(get_handler(i, name)))
+
+        # run a forward pass
+        pytorch_model(**first_batch)
+
+        # remove handles
+        for info in layer_info.values():
+            for handle in info["handles"]:
+                handle.remove()
+            del info["handles"]
+
+        # convert submodules to trt modules
+        logger.debug(f"Converting layers {min_layer} to {max_layer}...")
+        for layer_index, info in layer_info.items():
+            logger.debug(f"Converting layer {layer_index}...")
+            for name, shape in info["input_shapes"].items():
+                input = torch.zeros(shape, dtype=torch.float16, device=device)
+                # create trt module
+                trt_module = compile_trt_model(info["submodules"][name], input, batch_size, seqlen)
```

Comment:
>compile_trt_model [](http://example.com/codeflow?start=29&length=17)

should we check whether the input submodule is torch.nn.Linear before calling compile_trt_model? or the pass caller need make sure it?

Reply:
This is guaranteed by line 123. 
all of the sub modules we access here are found by that line where we specifically ask for NN.linear

Reply:
If user need to make sure the parent model, should we add document for what kinds of model types supports?

Reply:
or this pass only used for sparsegpt scenario?

Reply:
Yes, we are only using it for the sparsegpt scenario. I can add documentation in the pass docstring